public/js/gates.js

```diff
@@ -0,0 +1,223 @@
+	{ name: "Gate 78",  coordinates: [49.19811318,  -123.17304570], callsign: null, callsign: null, annotation: "International Flights | Heavy Gate" },
+	{ name: "Gate 79",  coordinates: [49.19770433,  -123.17307720], callsign: null, annotation: "USA Flights | Heavy Gate" },
+	{ name: "Gate 80",  coordinates: [49.19736210,  -123.17299700], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 81",  coordinates: [49.19713633,  -123.17315920], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 82",  coordinates: [49.19717426,  -123.17373681], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 83",  coordinates: [49.19711138,  -123.17433669], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 85",  coordinates: [49.19694580,  -123.17486553], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 86",  coordinates: [49.19679168,  -123.17541961], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 87",  coordinates: [49.19664818,  -123.17598470], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 88",  coordinates: [49.19641414,  -123.17645366], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 90",  coordinates: [49.19574250,  -123.17460362], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 91",  coordinates: [49.19588097,  -123.17420382], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 92",  coordinates: [49.19587412,  -123.17363526], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 93",  coordinates: [49.19587161,  -123.17306739], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 94",  coordinates: [49.19597194,  -123.17254916], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 95A", coordinates: [49.19545486,  -123.17278389], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 96",  coordinates: [49.19533293,  -123.17373550], callsign: null, annotation: "USA Flights" },
+	{ name: "Gate 96B", coordinates: [49.19531725,  -123.17293111], callsign: null, annotation: "USA Flights" }
+];
+
+async function fetchVatsimData() {
+    try {
+        const response = await fetch('https://data.vatsim.net/v3/vatsim-data.json');
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        throw new Error('Error fetching VATSIM data:', error);
+        alert('Error Fetching VATSIM data, please refresh this page');
+        throw error;
+    }
+}
+
+async function checkNearbyUsers(coordinates, vatsimData) {
+    try {
+        if (!vatsimData) return [];
+        const nearbyUsers = vatsimData.pilots.filter(pilot => {
+            const distance = getDistance(coordinates[0], coordinates[1], pilot.latitude, pilot.longitude);
+            return distance <= 0.03;
+        });
+        return nearbyUsers;
+    } catch (error) {
+        console.error('Error checking nearby aircraft:', error);
+        return [];
+    }
+}
+
+function getDistance(lat1, lon1, lat2, lon2) {
```

Comment:
This works, but I wonder about people's thoughts re: using an existing/maintained library for some of these calculations? Might make it more maintainable for someone coming along later (e.g. https://www.npmjs.com/package/geolib)

Reply:
Awesome idea! Currently unable to get it to work on my end. May require further tweaking to re-calibrate detection radius for stands.